library_generation/generate_composed_library.sh

```diff
@@ -0,0 +1,107 @@
+# pipe |. They key-value groups are in the form of `key=value` and will
+# be converted to an argument to generate_library.sh (`--key value`).
+#   example: "proto_path=google/cloud/asset/v1,destination_path=google-cloud-asset-v1-java,(...)|proto_path=google/cloud/asset/v1p2beta5,destination_path=google-cloud-asset-v1-java,(...)"
+#   In this case, generate_library.sh will be called once with
+#     generate_library.sh --proto_path google/cloud/asset/v1 --destination_path google-cloud-asset-v1-java
+#     and once with
+#     generate_library.sh --proto_path google/cloud/asset/v1p2bet5 --destination_path google-cloud-asset-v1p2beta5-java
+# --versions_file: a file of newline-separated version strings in the form "module:released-version:current-version". The versions will be applied to the pom.xml files and readmes
+#
+# Prerequisites
+# - Needs an `output` folder at the location of the calling shell. This folder
+# is automatically detected by `generate_library.sh` and this script ensures it
+# contains the necessary folders and files, specifically:
+#   - A "google" folder found in the googleapis repository
+#   - A "grafeas" folder found in the googleapis repository
+# Note: googleapis repo is found in https://github.com/googleapis/googleapis.
+
+set -xeo pipefail
+
+
+script_dir=$(dirname "$(readlink -f "$0")")
+library_generation_dir="${script_dir}"
+source "${script_dir}/utilities.sh"
+output_folder=$(get_output_folder)
+mkdir -p "${output_folder}"
+
+while [[ $# -gt 0 ]]; do
+key="$1"
+case $key in
+  -g|--generation_queries)
+    generation_queries="$2"
+    ;;
+  -r|--repository_path)
+    repository_path="$2"
+    ;;
+  -v|--versions_file)
+    versions_file="$2"
+    ;;
+  -p|--enable_postprocessing)
+    enable_postprocessing="$2"
+    ;;
+  *)
+    echo "Invalid option: [$1]"
+    exit 1
+    ;;
+esac
+shift
+shift # past argument or value
+done
+
+
+pushd "${output_folder}"
+if [[ "${repository_path}" == google-cloud-java/* ]]; then
+  echo 'this is a monorepo library'
+  library=$(echo "${repository_path}" | cut -d'/' -f2)
+  sparse_clone "https://github.com/googleapis/google-cloud-java.git" "${library} google-cloud-pom-parent google-cloud-jar-parent versions.txt .github"
+  versions_file="${versions_file:-"${output_folder}/google-cloud-java/versions.txt"}"
+else
+  echo 'this is a HW library'
+  git clone "https://github.com/googleapis/${repository_path}.git"
+  versions_file="${versions_file:-"${output_folder}/${repository_path}/versions.txt"}"
+fi
+
+owlbot_cli_source_folder=$(mktemp -d)
+popd # output_folder
+
+IFS="|"
+for query in $generation_queries; do
+  pushd "${output_folder}"
+  arguments=$(py_util get_generate_library_arguments "${query}")
+  arguments=$(py_util add_argument "${arguments}" "versions_file" "${versions_file}")
+  # we postprocess only once after all versions were processed
+  arguments=$(py_util add_argument "${arguments}" "enable_postprocessing" "false")
```

Comment:
Now that we completely split the responsibility of post-processing to a separate script, the parameter can probably be removed from generate_library.sh.

Reply:
Removed